pkg/urlutil/urlerrorhelper_test.go

```diff
@@ -0,0 +1,43 @@
+package urlutil
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+)
+
+func Test_urlparse01(t *testing.T) {
+	// error string as expected
+	errorString := `Get https://aaaaaaaa.blob.core.windows.net/nodeagentpackage-version9-0-0-381/Ubuntu-16.04/batch_config-ubuntu-16.04-1.5.9.tar.gz?sv=2018-03-28&sr=b&sig=a%secret%2Bsecret&st=2019-05-17T01%3A25%3A42Z&se=2021-05-24T01%3A25%3A42Z&sp=r: dial tcp 13.68.165.64:443: i/o timeout`
+	inputErr := fmt.Errorf("%s", errorString)
+	outputErr := RemoveUrlFromErr(inputErr)
+	if strings.Contains(outputErr.Error(), "https://") || !strings.Contains(outputErr.Error(), "[REDACTED]") {
```

Comment:
Might be good to assert that the secret value isn't present. A change could be introduced for example where [redacted] is present but the secret is also, and we should catch this.

Reply:
done